### PR TITLE
feat: Lazy loading | built Cart,Login,Signup pages

### PR DESCRIPTION
**Pull request by tushartiwari7**
Work done in this PR:
- Implemented Route based Code Splitting.
- improved User Stories (navigates to page where user wants to go after login page)
- Rewrote JSX for Cart , Signup and Login pages.

### CODE REVIEW

**Review by Bhakti199 — APPROVED** [review]
- This code looks fine to me.
- well written and structured.
- proper loading is done.
- This pr is good to merge.

File: src/pages/Wishlist/Wishlist.jsx [hunk 1]
```diff
@@ -7,21 +8,42 @@ export const Wishlist = () => {
...
-          wishlist.map((product) => (
-            <VerticalCard
-              key={product._id}
-              product={product}
-              isWishlistCard={true}
-            />
-          ))}
-      </div>
+    <main
+      className={`main flex flex-col ${!wishlist.length ? "flex-center" : ""}`}
```
Comment (paarthbajaj) [inline src/pages/Wishlist/Wishlist.jsx:12]: I liked this logic and I will try it too. Thanks

**Review by paarthbajaj — APPROVED** [review]
The logics you have used are really good and I will use them too. Implementation of loader is also good. You can merge this PR. Well done and goodluck!

File: src/pages/NotFound/NotFound.css [hunk 2]
```diff
@@ -0,0 +1,41 @@
+.gif {
+  max-height: 70vh;
+  object-fit: repeat;
+  width: 100%;
+}
+.giphy-embed {
+  position: absolute;
```
Comment (paarthbajaj) [inline src/pages/NotFound/NotFound.css]: You could use .pos-abs of your CL

File: src/pages/Wishlist/Wishlist.jsx [hunk 3]
```diff
@@ -36,7 +36,7 @@ export const Wishlist = () => {
           <p className="h2 ubuntu my-xs">Checkout our latest Furniture </p>
           <Link
             to="/products"
-            className="text-center full-width fs-l fw-lighter px-sm py-xs font-bebas cart-cta my-sm"
+            className="text-center full-width fs-l fw-lighter px-sm py-xs font-bebas btn-cta my-sm"
```
Comment (paarthbajaj) [inline src/pages/Wishlist/Wishlist.jsx:39]: Very good use of utility classes as well in all the commits.

File: src/pages/NotFound/NotFound.css [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+.grid2 {
+  grid-template-columns: 1fr 1fr;
+}
+.not-found-cta {
+  background: var(--text-color);
+  color: var(--bg-primary);
+  letter-spacing: 0.1ch;
+}
+.not-found-content {
+  width: 100%;
```
Comment (paarthbajaj) [inline src/pages/NotFound/NotFound.css:13]: You could use .full-width from your CL

**Pull request merged**